src/mlagility/common/filesystem.py

```diff
@@ -85,9 +94,11 @@ def _validate_script_in_database(self, script_name: str, func_name: str):
                 "that has not been added to the database yet."
             )
 
+    @dbtransaction
```

Comment:
Very cool, I've never implemented my own `@something` and its nice to see how it can be used to simplify code!

Reply:
Indeed very nice! Btw the term for it is "function decorators"

Reply:
Oh yeah! We saved a ton of lines with that!